cli/network.go

```diff
@@ -53,68 +53,14 @@ type nsListener struct {
 	cmd *exec.Cmd
 }
 
-func GetBridgeFromIndex(idx int) (string, string, error) {
-	var attr, bridge *netlink.LinkAttrs
-	var options string
-
-	links, err := netlink.LinkList()
-	if err != nil {
-		glog.Error(err)
-		return "", "", err
-	}
-
-	for _, link := range links {
-		if link.Type() != "veth" {
-			continue
-		}
-
-		if link.Attrs().Index == idx {
-			attr = link.Attrs()
-			break
-		}
-	}
-
-	if attr == nil {
-		return "", "", fmt.Errorf("cann't find nic whose ifindex is %d", idx)
-	}
-
-	for _, link := range links {
-		if link.Type() != "bridge" && link.Type() != "openvswitch" {
-			continue
-		}
-
-		if link.Attrs().Index == attr.MasterIndex {
-			bridge = link.Attrs()
-			break
-		}
-	}
-
-	if bridge == nil {
-		return "", "", fmt.Errorf("cann't find bridge contains nic whose ifindex is %d", idx)
+func createFakeBridge() {
+	// add an useless bridge to satisfy hypervisor, most of them need to join bridge.
+	la := netlink.NewLinkAttrs()
+	la.Name = fakeBridge
+	bridge := &netlink.Bridge{LinkAttrs: la}
+	if err := netlink.LinkAdd(bridge); err != nil && !os.IsExist(err) {
+		glog.Errorf("fail to create fake bridge: %v, %v", fakeBridge, err)
 	}
-
-	if bridge.Name == "ovs-system" {
```

Comment:
Why is this part removed? Is this code also a trick?

Reply:
Nop, the new design doesn't need to lookup bridge/ovs on host now